src/Data/Tree/AVL/Insertion.hs

```diff
@@ -59,70 +59,42 @@ insertZ k v = do
     goto (Plain k)             -- teleport to a key (or near it if absent)
     withLocus $ \case
       MLEmpty {} -> do
-        replaceWith =<< leaf k v minBound maxBound
-        return ()
+        replaceWith =<< leaf k v
 
-      MLLeaf {_mlKey, _mlPrevKey, _mlNextKey} -> do
-        let key0 = _mlKey
-            prev = _mlPrevKey
-            next = _mlNextKey
-        if Plain k == key0  -- update case, replace with new value
-        then
+      MLLeaf { _mlKey = key0 } -> do
+        if k == key0  -- update case, replace with new value
+        then do
             change $ do
                 here  <- use locus
                 here' <- setValue v here
                 locus .= here'
         else do
-            if Plain k `isInside` (prev, key0)
+            if k < key0
             then do
-                leaf0 <- leaf k v prev key0
-
-                splitInsertBefore leaf0
-
-                unless (prev == minBound) $ do
-                    goto prev
-                    change $ do
-                        here  <- use locus
-                        here' <- setNextKey (Plain k) here
-                        locus .= here'
+                splitInsertBefore =<< leaf k v
+                gotoPrevKey k
 
             else do
-                leaf0 <- leaf k v key0 next
+                splitInsertAfter =<< leaf k v
+                gotoNextKey k
 
-                splitInsertAfter leaf0
-
-                unless (next == maxBound) $ do
-                    goto next
-                    change $ do
-                        here  <- use locus
-                        here' <- setPrevKey (Plain k) here
-                        locus .= here'
-      _ -> error $ "insert: `goto k` ended in non-terminal node"
+      _ ->
+        error $ "insert: `goto k` ended in non-terminal node"
```

Comment:
oof, why newline and additional 2-space tab after `->`? one line was better by all means.

Reply:
Fix'd